common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -243,7 +242,8 @@ boolean availableToClaim() {
             if (state != STATE_AVAILABLE) {
                 return false;
             }
```

Comment:
We can also remove this if-condition, or change it to an assert. The handle should always be in the available state at this point, since it came off the local pool handles queue.

Reply:
@chrisvest emm, +1 for 'change it to an assert', @normanmaurer WDYT?

Reply:
sounds good